lib/dekorator.rb

```diff
@@ -22,9 +21,8 @@ class << self
       def decorate(object_or_collection, with: nil)
         return object_or_collection if decorable_object?(object_or_collection)
 
-        with = _guess_decorator(object_or_collection) if with.nil? && object_or_collection.present?
-
-        raise DecoratorNotFound, "Can't guess decorator for #{object_or_collection.class.name} object" if with.nil?
+        with ||= self if with != :__guess__ && self != Dekorator::Base
```

Comment:
```suggestion
        with = self if with != :__guess__ && self != Dekorator::Base
```

I think until `with == :__guess__` and current class is not `Dekorator::Base` we should use current class as decorator class. Even if `with` is precised.

`UserDecorator.decorate(post, with: PostDecorator)` or `UserDecorator.decorate(user, with: AdminDecorator)` doesn't make any sense.  
We should use `Dekorator::Base.decorate(record)` or  `Dekorator::Base.decorate(record, with: MyDecorator)`.

Reply:
Well, not sure since in a Rails application we have `ApplicationDecorator`.